lib/widgets/containers/tabs/dashboard_tab_widgets.dart

```diff
@@ -116,6 +133,155 @@ class PegOutViewModel extends BaseViewModel {
 
     await _router.pop();
   }
+
+  Future<void> estimateSidechainFee() async {
+    // who knows!
+    const kbyteInWithdrawal = 5;
+    final estimate = await _rpc.estimateSidechainFee(kbyteInWithdrawal);
+    sidechainFee = estimate;
+  }
+
+  void onPegOut(BuildContext context) async {
+    if (pegOutAmount == null) {
+      log.e('withdrawal amount was empty');
+      return;
+    }
+    if (sidechainFee == null) {
+      log.e('sidechain fee was empty');
+      return;
+    }
+
+    // 1. Get refund address for the sidechain withdrawal. This can be any address we control on the SC.
+    final refund = await _rpc.getRefundAddress();
+    log.d('got refund address: $refund');
+
+    // 2. Get MC fee
+    // This happens with the `getaveragemainchainfees` RPC. This
+    // is passed straight on to the mainchain `getaveragefee`,
+    // which is added in the Drivechain implementation of Bitcoin
+    // Core.
+    // This, as opposed to `estimatesmartfee`, is used is because
+    // we need everyone to get the same results for withdrawal
+    // bundle validation.
+    //
+    // The above might not actually be correct... What's the best
+    // way of doing this?
```

Comment:
This comment is a leftover from my old code, but then I moved the code it was supposed to be related to, without deleting the comment. Rip it out!